src/H5Oflush.c

```diff
@@ -256,6 +259,10 @@ H5O_refresh_metadata(H5O_loc_t *oloc, hid_t oid)
     if (objs_incr && file)
         H5F_decr_nopen_objs(file);
 
+    if (vol_obj && vol_obj->connector)
```

Comment:
This seems overly complex.  Just decrement the connector's refcount before throwing an error, for the two places where it matters.